provider/workspace_preset.go

```diff
@@ -93,6 +96,29 @@ func workspacePresetDataSource() *schema.Resource {
 				Required:     true,
 				ValidateFunc: validation.StringIsNotEmpty,
 			},
+			"description": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Describe what this preset does.",
+			},
+			"icon": {
+				Type: schema.TypeString,
+				Description: "A URL to an icon that will display in the dashboard. View built-in " +
+					"icons [here](https://github.com/coder/coder/tree/main/site/static/icon). Use a " +
+					"built-in icon with `\"${data.coder_workspace.me.access_url}/icon/<path>\"`.",
+				ForceNew: true,
+				Optional: true,
+				ValidateFunc: func(value interface{}, label string) ([]string, []error) {
```

Comment:
TODO: use the helper ValidateURL function from https://github.com/coder/terraform-provider-coder/pull/421/files#diff-9c5c4a2dd840a61a4b9b772d602c83bd6a692b6e1e7a4a850fc30717f3ee67a8R10